internal/command/jobs/printers.go

```diff
@@ -0,0 +1,250 @@
+package jobs
+
+import (
+	"fmt"
+	"io"
+	"sort"
+	"text/tabwriter"
+	"time"
+
+	"github.com/signadot/cli/internal/config"
+	"github.com/signadot/cli/internal/sdtab"
+	"github.com/signadot/go-sdk/client/artifacts"
+	"github.com/signadot/go-sdk/models"
+	"github.com/xeonx/timeago"
+)
+
+const MaxJobListing = 20
+
+type jobRow struct {
+	Name        string `sdtab:"NAME"`
+	Environment string `sdtab:"ENVIRONMENT"`
+	CreatedAt   string `sdtab:"CREATED AT"`
+	StartedAt   string `sdtab:"STARTED AT"`
+	Duration    string `sdtab:"DURATION"`
+	Status      string `sdtab:"STATUS"`
+}
+
+func printJobTable(cfg *config.JobList, out io.Writer, jobs []*models.Job) error {
+	t := sdtab.New[jobRow](out)
+	t.AddHeader()
+
+	sort.Slice(jobs, func(i, j int) bool {
+		if jobs[i].Status.Attempts[0].Phase == "queued" {
+			return true
+		}
+
+		if jobs[j].Status.Attempts[0].Phase == "queued" {
+			return false
+		}
+
+		t1, err1 := time.Parse(time.RFC3339, jobs[i].CreatedAt)
+		t2, err2 := time.Parse(time.RFC3339, jobs[j].CreatedAt)
+		if err1 != nil || err2 != nil {
+			return false
+		}
+
+		return t2.Before(t1)
+	})
+
+	counter := 0
+	for _, job := range jobs {
+		if counter == MaxJobListing {
+			break
+		}
+
+		switch {
+		case cfg.ShowCompleted:
+		case !cfg.ShowCompleted && job.Status.Phase != "completed":
+		default:
+			continue
+		}
+
+		counter += 1
+
+		createdAt, duration := getCreatedAtAndDuration(job)
+
+		environment := ""
+		routingContext := job.Spec.RoutingContext
+		switch {
+		case routingContext == nil:
+		case len(routingContext.Sandbox) > 0:
+			environment = fmt.Sprintf("sandbox=%s", routingContext.Sandbox)
+		case len(routingContext.Routegroup) > 0:
+			environment += fmt.Sprintf("routegroup=%s", routingContext.Routegroup)
+		}
+
+		t.AddRow(jobRow{
+			Name:        job.Name,
+			Environment: environment,
+			StartedAt:   createdAt,
+			Duration:    duration,
+			Status:      job.Status.Phase,
+			CreatedAt:   getCreatedAt(job),
+		})
+	}
+	return t.Flush()
+}
+
+func printJobDetails(cfg *config.Job, out io.Writer, job *models.Job) error {
+	tw := tabwriter.NewWriter(out, 0, 0, 3, ' ', 0)
+
+	createdAt, duration := getCreatedAtAndDuration(job)
+
+	fmt.Fprintf(tw, "Job Name:\t%s\n", job.Name)
+	fmt.Fprintf(tw, "Status:\t%s\n", job.Status.Phase)
+	fmt.Fprintf(tw, "Environment:\t%s\n", getJobEnvironment(job))
+	fmt.Fprintf(tw, "Created At:\t%s\n", getCreatedAt(job))
+	fmt.Fprintf(tw, "Started At:\t%s\n", createdAt)
+	fmt.Fprintf(tw, "Duration:\t%s\n", duration)
```

Comment:
Duration doesn't seem to be available until the job is complete. So maybe hide it when it's not set? Same for the other properties (eg. startTime).

<img width="629" alt="image" src="https://github.com/signadot/cli/assets/3791487/571d5851-33a0-4a95-9683-51667cb76b4c">